tests/unit/settled-test.js

```diff
@@ -144,6 +145,32 @@ module('settled', function(hooks) {
 
       assert.strictEqual(isSettled(), true, 'post cond');
     });
+
+    test('when router is transitioning', async function(assert) {
+      assert.expect(3);
+
+      setContext(this);
+      assert.strictEqual(isSettled(), true);
+
+      this.owner = {
```

Comment:
I'd rather not use mocking style unit tests like this. In past versions of ember-test-helpers these style tests led to brittleness across Ember versions (and false confidence about what works in reality).